packages/twenty-server/src/modules/workflow/workflow-trigger/workspace-services/workflow-trigger.workspace-service.ts

```diff
@@ -203,6 +203,10 @@ export class WorkflowTriggerWorkspaceService {
         workflowVersionNullable,
       );
 
+    if (workflowVersion.status !== WorkflowVersionStatus.ACTIVE) {
+      return;
+    }
```

Comment:
When the user wants to activate a workflow version, it might not be necessary to deactivate the previous version as it might already have.

If the user deactivates an active workflow, then creates a draft and wants to activate that draft, the `performDeactivationSteps` function should be short-circuited as it doesn't have to perform these deactivation steps on a deactivated version. You can see the error happening in the video below.

Did I miss something?

https://github.com/user-attachments/assets/baff8ab4-ca50-4884-afe9-c2243184eaba

Reply:
Got it now 👌